src/main/java/org/hibernate/infra/bot/EditPullRequestBodyAddIssueLinks.java

```diff
@@ -0,0 +1,155 @@
+import org.kohsuke.github.GHIssueState;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHPullRequestCommitDetail;
+import org.kohsuke.github.GHRepository;
+
+/**
+ * @author Marco Belladelli
+ */
+public class EditPullRequestBodyAddIssueLinks {
+	private static final Logger LOG = Logger.getLogger( EditPullRequestBodyAddIssueLinks.class );
+
+	private static final String START_MARKER = "<!-- Hibernate GitHub Bot issue links start -->";
+
+	private static final String END_MARKER = "<!-- Hibernate GitHub Bot issue links end -->";
+
+	private static final String LINK_TEMPLATE = "https://hibernate.atlassian.net/browse/%s";
+
+	@Inject
+	DeploymentConfig deploymentConfig;
+
+	void pullRequestChanged(
+			@PullRequest.Opened @PullRequest.Reopened @PullRequest.Edited @PullRequest.Synchronize
+			GHEventPayload.PullRequest payload,
+			@ConfigFile( "hibernate-github-bot.yml" ) RepositoryConfig repositoryConfig) throws IOException {
+		editPullRequestBodyAddIssueLinks( payload.getRepository(), repositoryConfig, payload.getPullRequest() );
+	}
+
+	void checkRunRequested(
+			@CheckRun.Rerequested GHEventPayload.CheckRun payload,
+			@ConfigFile( "hibernate-github-bot.yml" ) RepositoryConfig repositoryConfig) throws IOException {
+		for ( GHPullRequest pullRequest : payload.getCheckRun().getPullRequests() ) {
+			editPullRequestBodyAddIssueLinks( payload.getRepository(), repositoryConfig, pullRequest );
+		}
+	}
+
+	void checkSuiteRequested(
+			@CheckSuite.Requested @CheckSuite.Rerequested GHEventPayload.CheckSuite payload,
+			@ConfigFile( "hibernate-github-bot.yml" ) RepositoryConfig repositoryConfig) throws IOException {
+		for ( GHPullRequest pullRequest : payload.getCheckSuite().getPullRequests() ) {
+			editPullRequestBodyAddIssueLinks( payload.getRepository(), repositoryConfig, pullRequest );
+		}
+	}
+
+	private void editPullRequestBodyAddIssueLinks(
+			GHRepository repository,
+			RepositoryConfig repositoryConfig,
+			GHPullRequest pullRequest) throws IOException {
+		if ( !shouldCheck( repository, pullRequest ) ) {
+			return;
+		}
+
+		if ( repositoryConfig == null || repositoryConfig.enableEditing == null || repositoryConfig.enableEditing.equals( Boolean.FALSE )
+				|| repositoryConfig.jira == null || repositoryConfig.jira.getIssueKeyPattern().isEmpty() ) {
+			return;
+		}
+
+		final Set<String> issueKeys = new HashSet<>();
+		// Collect all issue keys from commit messages
+		repositoryConfig.jira.getIssueKeyPattern().ifPresent( issueKeyPattern -> {
+			for ( GHPullRequestCommitDetail commitDetails : pullRequest.listCommits() ) {
+				final GHPullRequestCommitDetail.Commit commit = commitDetails.getCommit();
+				final List<String> commitIssueKeys = CommitMessages.extractIssueKeys(
+						issueKeyPattern,
+						commit.getMessage()
+				);
+				issueKeys.addAll( commitIssueKeys );
+			}
+		} );
+
+		if ( issueKeys.isEmpty() ) {
+			LOG.info( "Found no issue keys in commits, terminating." );
```

Comment:
Without context, this is probably only useful while testing.
```suggestion
			LOG.debug( "Found no issue keys in commits, terminating." );
```